x-pack/plugins/ingest_manager/common/services/is_agent_upgradeable.ts

```diff
@@ -14,9 +14,12 @@ export function isAgentUpgradeable(agent: Agent, kibanaVersion: string) {
     return false;
   }
   if (agent.unenrollment_started_at || agent.unenrolled_at) return false;
-  const kibanaVersionParsed = semver.parse(kibanaVersion);
-  const agentVersionParsed = semver.parse(agentVersion);
-  if (!agentVersionParsed || !kibanaVersionParsed) return false;
   if (!agent.local_metadata.elastic.agent.upgradeable) return false;
-  return semver.lt(agentVersionParsed, kibanaVersionParsed);
+
+  // make sure versions are only the number before comparison
+  const agentVersionNumber = semver.coerce(agentVersion);
+  if (!agentVersionNumber) throw new Error('agent version is invalid');
+  const kibanaVersionNumber = semver.coerce(kibanaVersion);
+  if (!kibanaVersionNumber) throw new Error('kibana version is invalid');
+  return semver.lt(agentVersionNumber, kibanaVersionNumber);
```

Comment:
nitpick: I think this is doing the same not sure which approach we should favorise https://github.com/elastic/kibana/blob/master/x-pack/plugins/ingest_manager/server/services/agents/checkin/state_new_actions.ts/#L127-L137

Reply:
The thing that makes me a bit paranoid is `semver.lt('7.9.0-SNAPSHOT', '7.9.0', { includePrerelease: true });` or semver.lt('7.9.0-SNAPSHOT', '7.9.0', { includePrerelease: false });` evaluates to being `true`, and in our case because a version is a snapshot, it shouldn't be considered less than when the versions are the same.